### PR TITLE
Domain averaged quantities in 2D and 3D and protected parameters in 2D

### PR DESCRIPTION
**Pull request by matt-frey**
In the PS3D paper we calculated the domain-averaged kinetic energy and enstrophy. I suggest we directly calculate the domain-averaged quantities in EPIC.

### CODE REVIEW

File: src/2d/fields/field_diagnostics_netcdf.f90 [hunk 1]
```diff
@@ -140,9 +140,9 @@ subroutine create_netcdf_field_stats_file(basename, overwrite, l_restart)
             call define_netcdf_dataset(                                     &
                 ncid=ncid,                                                  &
                 name='ke',                                                  &
-                long_name='kinetic energy',                                 &
+                long_name='domain-avareged kinetic energy',                 &
```
Comment (sjboeing) [inline src/2d/fields/field_diagnostics_netcdf.f90:143]: minor typo in "averaged"

File: src/2d/utils/parameters.f90 [hunk 2]
```diff
@@ -109,4 +118,24 @@ subroutine update_parameters
         max_num_parcels = int(nx * nz * parcel%min_vratio * parcel%size_factor)
 
     end subroutine update_parameters
+
+        subroutine set_mesh_spacing(ext, nc)
```
Comment (sjboeing) [inline src/2d/utils/parameters.f90:122]: Change indentation?

**Review by sjboeing — APPROVED** [review]
Looks good to me, Matthias. Thanks for working on this.

File: src/2d/utils/parameters.f90 (hunk 2 above)
Comment (matt-frey) [reply src/2d/utils/parameters.f90:122]: Oh, yes, thanks.

File: src/2d/fields/field_diagnostics_netcdf.f90 (hunk 1 above)
Comment (matt-frey) [reply src/2d/fields/field_diagnostics_netcdf.f90]: Thanks.

**Pull request merged**